src/Orleans.Runtime/Streams/QueueBalancer/LeaseBasedQueueBalancer.cs

```diff
@@ -419,43 +379,52 @@ private async Task ScheduleUpdateResponsibilities(HashSet<SiloAddress> activeSil
             }
         }
 
+        /// <summary>
+        /// Greedy silos
+        /// </summary>
+        /// <param name="overflow"></param>
+        /// <param name="activeSilos"></param>
+        /// <returns></returns>
+        private bool AmGreedy(int overflow, HashSet<SiloAddress> activeSilos)
+        {
+            // If using multiple stream providers, this will select the same silos to be greedy for
+            //   all providers, aggravating inbalance as stream provider count increases.
```

Comment:
In practice I don't think this will be an issue, but I'm glad you highlighted it

Reply:
Understood.  A little history:  This is an issue in an existing service which has one or more stream providers per tenant.  Addressing it was a specific ask by the team when working out the requirements of the lease base queue balancer.  This note is primarily to help track this shortcoming, should future maintainers need address it.